### PR TITLE
Dev/lua

### PR DESCRIPTION
**Pull request by jelavice**

### CODE REVIEW

File: open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp
Comment (nubertj) [inline open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp:39]: comments

File: open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp
Comment (nubertj) [inline open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp:1]: maybe move to a "legacy" folder or so?

**Review by nubertj — CHANGES_REQUESTED** [review]

**Comment by nubertj** [comment]
I think overall the new structure is cool. But before merging it should be cleaned up and maybe structured a bit more nicely. Also I could not compile it although installing lua 5.2. I added this information to the readme. Anything else required?

**Comment by nubertj** [comment]
okay, now i could run it and it works :) good to merge after the cleanup.

File: open3d_slam/include/open3d_slam/Parameters.hpp
Comment (jelavice) [reply open3d_slam/include/open3d_slam/Parameters.hpp:130]: yes, I would put formatting in a separate PR

File: open3d_slam_io/lua_parameter_dictionary/src/example.cpp
Comment (jelavice) [reply open3d_slam_io/lua_parameter_dictionary/src/example.cpp:30]: This is a minimal exapmle just to run the dictionary
I thought it would be helpful to have something like this

File: open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp
Comment (jelavice) [reply open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp:1]: legacy folder or just delete it?

File: open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp
Comment (jelavice) [reply open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp:19]: just to show how to load parameters, but maybe it is redundant since this is already done in the mapping node. Should I remove it?

File: open3d_slam_io/lua_parameter_dictionary/src/example.cpp
Comment (nubertj) [reply open3d_slam_io/lua_parameter_dictionary/src/example.cpp:30]: yeah might be useful. Just thought regarding the double_shizzle ;)

File: open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp
Comment (nubertj) [reply open3d_slam_io/open3d_slam_lua_io/src/minimal_example.cpp:19]: i think both is fine. But sometimes less is more

File: open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp
Comment (nubertj) [reply open3d_slam_io/open3d_slam_yaml_io/include/open3d_slam_yaml_io/parameter_loaders.hpp:1]: also good :)

**Review by nubertj — APPROVED** [review]

**Pull request merged**
